packages/headless/src/ssr-next/commerce/types/build.ts

```diff
@@ -24,9 +25,62 @@ export interface SearchBuildConfig extends CommonBuildConfig {
   >['parameters'];
 }
 
-export interface RecommendationBuildConfig extends CommonBuildConfig {
-  recommendations?: []; // TODO: KIT-4619
-}
+/**
+ * RecommendationBuildConfig defines the shape of the config object required when fetching static state for recommendations.
+ *
+ * The `recommendations` property is a string array containing the names of the recommendation controllers
+ * that were defined in your engine definition. This allows you to specify which recommendation controllers
+ * should be included in the SSR request.
+ *
+ * For example, given the following engine definition:
+ *
+ * ```ts
+ * const {recommendationEngineDefinition} = defineCommerceEngine({
+ *   controllers: {
+ *     popularViewed: defineRecommendations({options: {slotId: '...'}}),
+ *     popularBought: defineRecommendations({options: {slotId: '...'}}),
+ *     viewedTogether: defineRecommendations({options: {slotId: '...'}}),
+ *     // ...other controllers
+ *   },
+ * });
+ *
+ * // When fetching static state for recommendations:
+ * const staticState = await recommendationEngineDefinition.fetchStaticState({
+ *   recommendations: ['popularBought', 'viewedTogether'],
+ *   // ...other config
+ * });
+ * ```
+ *
+ * In this example, only the `popularBought` and `viewedTogether` recommendation controllers will be included in the SSR request.
```

Comment:
We could add an explanation about why someone would want this, and why it's necessary.

> For example, on a storefront homepage, it might make sense to display a carousel of popular products. However, showing recommendations based on items frequently bought together may not be relevant in that context and should not be retrieved unnecessarily.